src/main/kotlin/christmas/domain/order/Reservation.kt

```diff
@@ -0,0 +1,38 @@
+package christmas.domain.order
+
+import christmas.domain.menu.Category
+import christmas.domain.menu.Money
+import java.time.DayOfWeek
+import java.time.LocalDate
+
+data class Reservation(val date: LocalDate, val order: Order) {
+
+    init {
+        val after = LocalDate.of(2023, 11, 30)
+        val before = LocalDate.of(2024, 1, 1)
+        require(date.isAfter(after) && date.isBefore(before)) {
+            "valid date must be between $after and $before"
+        }
+    }
```

Comment:
```suggestion
    init {
        val firstDay = LocalDate.of(2024, 12, 1)
        val lastDay = LocalDate.of(2024, 12, 31)
        require(date in firstDay..lastDay) {
            "valid date must be from $firstDay to $lastDay"
        }
    }
```

This one might be more readable!